bodhi/tests/server/test_util.py

```diff
@@ -247,6 +247,16 @@ def test_failed(self):
                          "<span class='label label-danger'>Failed</span>")
 
 
+class TestCanWaiveTestResults(base.BaseTestCase):
+    """Test the can_waive_test_results() function."""
+    @mock.patch.dict('bodhi.server.util.config', {'test_gating.required': True})
+    def test_can_waive_test_results(self):
+        u = Update.query.all()[0]
+        u.test_gating_status = TestGatingStatus.failed
+        u.status = models.UpdateStatus.testing
+        self.assertTrue(util.can_waive_test_results(None, u))
```

Comment:
We should add a few more tests to hit the other branches of the if statement:

1. It should return ```False``` if the gating is disabled, even if other conditions are met.
2. It should return ```False``` if the tests passed, even if other conditions are met.
3. It should return ```False``` if the update is stable.